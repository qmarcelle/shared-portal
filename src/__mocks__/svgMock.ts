import React from 'react';

const SvgMock = React.forwardRef<
  HTMLSpanElement,
  React.SVGProps<SVGSVGElement>
>((props, ref) => {
  return React.createElement('span', {
    ref,
    ...props,
    'data-testid': 'svg-mock',
  });
});

SvgMock.displayName = 'SvgMock';

export default SvgMock;
export const ReactComponent = SvgMock;
