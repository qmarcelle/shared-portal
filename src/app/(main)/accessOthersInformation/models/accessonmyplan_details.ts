export interface AccessOnMyPlanDetails {
  memberName: string;
  DOB: string;
  isOnline: boolean;
}
