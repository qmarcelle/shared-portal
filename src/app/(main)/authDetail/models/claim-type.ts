export enum ClaimType {
  Medical = 'Medical',
  Pharmacy = 'Pharmacy',
  Dental = 'Dental',
}
