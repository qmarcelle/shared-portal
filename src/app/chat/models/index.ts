// Export all chat-related types
export type {
  CobrowseConfig,
  CobrowseInitResponse,
  CobrowseSession,
  CobrowseSessionResponse,
} from './cobrowse';

// Make sure the ChatMessage type is consistent by defining it inline here
export interface ChatMessage {
  id: string;
  text: string;
  sender: 'user' | 'agent' | 'bot';
  timestamp: Date;
}

export * from './plans';
export type {
  ChatEligibility,
  ChatPayload,
  ChatSession,
  ChatSessionJWT,
  ChatState,
  PlanSwitcherState,
} from './session';
export type {
  BusinessDay,
  BusinessHours,
  ChatConfig,
  ChatOption,
  CobrowseState,
  UserEligibility,
} from './types';
