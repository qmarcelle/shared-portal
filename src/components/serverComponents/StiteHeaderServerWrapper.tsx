import { getPolicyInfo } from '@/actions/getPolicyInfo';
import { getUserProfiles } from '@/actions/profileData';
import { auth } from '@/auth';
import { logger } from '@/utils/logger';
import { transformPolicyToPlans } from '@/utils/policy_computer';
import SiteHeader from '../foundation/SiteHeader';
import SiteHeaderPBEError from '../foundation/SiteHeaderPBEError';

export const SiteHeaderServerWrapper = async () => {
  //const visibityRules = await getVisibilityRules();
  const session = await auth();
  const visibityRules = session?.user.vRules;
  if (visibityRules) {
    try {
      const profiles = await getUserProfiles();
      const selectedProfile = profiles.find((item) => item.selected == true)!;
      const selectedPlanId = selectedProfile?.plans.find(
        (item) => item.selected == true,
      )?.memCK;
      let policyInfo;
      try {
        policyInfo =
          selectedProfile!.plans?.length > 0
            ? await getPolicyInfo(
                selectedProfile!.plans.map((item) => item.memCK),
              )
            : null;
      } catch (error) {
        logger.error('SiteHeader Action - getPolicyInfo Failed {}', error);
      }
      const plans = policyInfo ? transformPolicyToPlans(policyInfo) : [];
      const selectedPlan = selectedPlanId
        ? plans?.find((item) => item.memeCk == selectedPlanId)
        : undefined;

      return (
        <SiteHeader
          visibilityRules={visibityRules}
          profiles={profiles}
          plans={plans}
          selectedPlan={selectedPlan}
          selectedProfile={selectedProfile}
          userId={session?.user.id}
          groupId={session?.user.currUsr.plan?.grpId}
        />
      );
    } catch (error) {
      logger.error('SiteHeader Action - getPBEConsentDetails Failed {}', error);
      return <SiteHeaderPBEError />;
    }
  } else {
    return <></>;
  }
};
