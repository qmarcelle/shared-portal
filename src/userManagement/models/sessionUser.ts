import { VisibilityRules } from '@/visibilityEngine/rules';

export enum UserRole {
  MEMBER = 'MEM',
  PERSONAL_REP = 'PR',
  AUTHORIZED_USER = 'AU',
  NON_MEM = 'NM',
}

export type SessionUser = {
  id: string;
  currUsr: {
    umpi: string;
    fhirId: string;
    role: UserRole;
    plan:
      | {
          memCk: string;
          sbsbCk: string;
          grgrCk: string;
          grpId: string;
          subId: string;
          fhirId: string;
          ntwkId: string;
        }
      | undefined;
  };
  rules?: string;
  vRules?: VisibilityRules;
  subscriberId?: string;
};
