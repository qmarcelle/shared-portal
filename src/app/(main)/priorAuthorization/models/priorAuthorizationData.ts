import { ClaimDetails } from '../../../../models/claim_details';

export const priorAuthorizationData: ClaimDetails[] = [
  {
    id: 'Claim1',
    claimStatus: 'Pending',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Physical Therapy',
    memberName: 'Chris Hall',
    serviceDate: '04/06/23',
    totalBilled: '',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'Anand Patel',
    ReferredTo: 'Anand Patel',
    priorAuthFlag: true,
  },
  {
    id: 'Claim2',
    claimStatus: 'Approved',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Magnetic Resonance Imaging (MRI)',
    memberName: 'Chris Hall',
    serviceDate: '01/23/23',
    totalBilled: '145.00',
    planPaid: '114.76',
    myShare: '30.24',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
  {
    id: 'Claim3',
    claimStatus: 'Partial Approval',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Continuous Positive Airway Pressure Machiâ€¦',
    memberName: 'Chris Hall',
    serviceDate: '08/10/23',
    totalBilled: '403.98',
    planPaid: '338.37',
    myShare: '65.61',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'Anand Patel',
    priorAuthFlag: true,
  },
  {
    id: 'Claim4',
    claimStatus: 'Denied',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Home Health Services',
    memberName: 'Chris Hall',
    serviceDate: '07/05/23',
    totalBilled: '403.98',
    planPaid: '0.00',
    myShare: '403.98',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'Home Health Services',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
  {
    id: 'Claim5',
    claimStatus: 'Approved',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Brittant Johnson',
    memberName: 'Chris Hall',
    serviceDate: '07/04/23',
    totalBilled: '183.00',
    planPaid: '150.85',
    myShare: '32.15',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
];
