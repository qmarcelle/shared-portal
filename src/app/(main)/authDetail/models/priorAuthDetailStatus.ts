import { ClaimStatus } from '@/app/(main)/authDetail/models/claim-status';
import { ClaimType } from '@/app/(main)/authDetail/models/claim-type';

export interface PriorAuthDetailsStatus {
  priorAuthDetailType: ClaimType;
  priorAuthDetailName: string;
  dateOfVisit: string;
  priorAuthDetailStatus: ClaimStatus;
  member: string;
  PriorAuthReferenceId: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  authInfo: any;
  referredName: string;
}
