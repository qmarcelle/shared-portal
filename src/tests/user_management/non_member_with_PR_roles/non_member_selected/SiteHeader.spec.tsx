import { SiteHeaderServerWrapper } from '@/components/serverComponents/StiteHeaderServerWrapper';
import { mockedAxios } from '@/tests/__mocks__/axios';
import { UserRole } from '@/userManagement/models/sessionUser';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

// PBE Call
mockedAxios.get.mockResolvedValueOnce({
  data: {
    data: {
      getPBEmessage: 'Person Record fetched Successfully ',
      getConsentmessage: 'Person Record fetched Successfully ',
      getPBECorrelationId: '0124b43b-efd0-42bc-b904-cb2ac9c712b9',
      consentCorrelationId: 'cd30f2b3-3bae-42f3-b93a-8fa42f3884e7',
      getPBEDetails: [
        {
          umpid: '57c85test3ebd23c7db88245',
          userName: 'Testuser553',
          personFHIRID: '30345928-abcd-ef01-2345-6789abcdef51',
          firstName: 'Alpha',
          middleName: 'S',
          lastName: 'Beta',
          suffix: 'Mr.',
          address1: 'street1',
          address2: 'street2',
          phoneNumber: '8095469997',
          city: 'Ongole',
          state: 'AP',
          zip: 34566,
          dob: '2019-10-29T00:00:10.0000000+00:00',
          gender: 'M',
          email: 'sub.m@gmail.com',
          relationshipInfo: [
            {
              personRoleType: 'PR',
              org: 'bcbst_facets',
              roleTermDate: '2030-11-30T00:00:00.0000000+00:00',
              nativeId: '38922455201-100000',
              primaryPlanFlag: false,
              patientFHIRID: '30345928-abcd-ef01-2345-6789abcdef52',
              userName: 'Testuser553',
              memeCk: '3349134',
              clientId: '194',
              multiPlanConfirmed: false,
              multiPlanConfirmedDate: '2030-11-30T00:00:00.0000000+00:00',
              approvalRequestId: '38922455201-1123456787',
              relatedPersons: [
                {
                  relatedPersonUMPID: '57c85test3ebd23c7db88244',
                  relatedPersonFirstName: 'AMDS',
                  relatedPersonLastName: 'Deped',
                  relatedPersonMiddleName: 'S',
                  relatedPersonSuffix: 'Mr.',
                  relatedPersonNativeId: '38922455200-100000',
                  relatedPersonFHIRID: '',
                  relatedPersonPatientFHIRID: '',
                  relatedPersonRelationshipTermDate:
                    '2030-11-30T00:00:00.0000000+00:00',
                  relatedPersonRoleType: 'Subscriber',
                  relatedPersonDob: '2030-11-30T00:00:00.0000000+00:00',
                  relatedPersonApprovalRequestId: '',
                  relatedPersonMemeCk: '6765458',
                },
              ],
            },
            {
              personRoleType: 'PR',
              org: 'bcbst_facets',
              roleTermDate: '2030-11-30T00:00:00.0000000+00:00',
              nativeId: '38922455201-100000',
              primaryPlanFlag: false,
              patientFHIRID: '30345928-abcd-ef01-2345-6789abcdef52',
              userName: 'Testuser553',
              memeCk: '3349134',
              clientId: '194',
              multiPlanConfirmed: false,
              multiPlanConfirmedDate: '2030-11-30T00:00:00.0000000+00:00',
              approvalRequestId: '38922455201-1123456787',
              relatedPersons: [
                {
                  relatedPersonUMPID: '57c85test3ebd23c7db88287',
                  relatedPersonFirstName: 'Robert',
                  relatedPersonLastName: 'Cook',
                  relatedPersonMiddleName: 'S',
                  relatedPersonSuffix: 'Mr.',
                  relatedPersonNativeId: '38922455200-100000',
                  relatedPersonFHIRID: '',
                  relatedPersonPatientFHIRID: '',
                  relatedPersonRelationshipTermDate:
                    '2030-11-30T00:00:00.0000000+00:00',
                  relatedPersonRoleType: 'Subscriber',
                  relatedPersonDob: '2030-11-30T00:00:00.0000000+00:00',
                  relatedPersonApprovalRequestId: '',
                  relatedPersonMemeCk: '6765458',
                },
              ],
            },
          ],
        },
      ],
      consentDetails: [
        {
          id: 'e7d4e7e9-5f98-4e35-9141-de8c8eb944a6',
          name: 'Consent - Limited Access',
          description:
            'This consent is governed by limited access policy for HIV and Cancer information',
          status: 'active',
          performer: 'john.doe@bcbst.com',
          requester: 'Person/24baffdb-76b7-4cb1-bf97-47f4bcf5574a',
          requestee: 'Patient/e3c212e5-1b7f-4521-8ba1-3b48befa98d6',
          policyId: 'ce3ee5b4-3810-4c8a-bfd9-9d8681853a7e',
          policyBusinessIdentifier: 'pol_10237_lim_ac',
          type: 'PEOPLE_TO_PEOPLE',
          effectiveOn: '2023-01-01T12:00:00Z',
          expiresOn: '2023-12-31T12:00:00Z',
          signedOn: '2023-01-01T12:00:00Z',
          modifiedAt: '2023-01-01T12:00:00Z',
          accessControl: 'PERMIT_ALL',
          options: [
            {
              id: 'f83a70bf-736d-4dfa-96eb-31d5c5a09535',
              businessIdentifier: 'option_biz_id_001',
              caption: 'HIV',
              mappings: [
                {
                  criteria: 'CODE',
                  key: 'http://hl7.org/fhir/sid/icd-10',
                  value: 'B20,B9735',
                },
              ],
            },
          ],
        },
      ],
    },
    details: {
      componentName: 'string',
      componentStatus: 'string',
      returnCode: 'string',
      subSystemName: 'string',
      message: 'string',
      problemTypes: 'string',
      innerDetails: {
        statusDetails: [
          {
            componentName: 'string',
            componentStatus: 'string',
            returnCode: 'string',
            subSystemName: 'string',
            message: 'string',
            problemTypes: 'string',
            innerDetails: {
              statusDetails: ['string'],
            },
          },
        ],
      },
    },
  },
});

// Policy Info Call

jest.mock('src/auth', () => ({
  auth: jest.fn(() =>
    Promise.resolve({
      user: {
        id: 'testUser',
        currUsr: {
          umpi: '57c85test3ebd23c7db88245',
          role: UserRole.NON_MEM,
          plan: undefined,
        },
        vRules: {},
      },
    }),
  ),
}));

describe('SiteHeader for Non Member with PR Role', () => {
  let containerSnap: HTMLElement;
  beforeAll(async () => {
    const SiteHeader = await SiteHeaderServerWrapper();
    const { container } = render(SiteHeader);
    containerSnap = container;
  });

  it('should render UI correctly with no nav menu, plan selector and action buttons', () => {
    expect(screen.getByText('Alpha Beta')).toBeVisible();
    expect(screen.queryByText('View Plan:')).toBeNull();
    expect(mockedAxios.get).toHaveBeenCalledWith(
      '/searchMemberLookupDetails/getPBEConsentDetails',
      { params: { isPBERequired: true, userName: 'testUser' } },
    );
    expect(containerSnap).toMatchSnapshot();
  });
});
