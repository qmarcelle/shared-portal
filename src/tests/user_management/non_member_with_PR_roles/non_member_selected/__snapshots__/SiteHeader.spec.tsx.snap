// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteHeader for Non Member with PR Role should render UI correctly with no nav menu, plan selector and action buttons 1`] = `
<div>
  <nav
    class="primary-color sm:pt-[74px] lg:pt-[74px]"
  >
    <div
      class="h-18 w-full fixed top-0 left-0 right-0 flex justify-between border-b bg-white z-50"
    >
      <div
        class="flex items-center"
      >
        <div
          class="flex lg:hidden h-18 w-18 items-center justify-center border-r"
        >
          <button
            aria-controls="menu-bar"
            aria-expanded="false"
            class="p-0 justify-center"
            data-collapse-toggle="menu-bar"
            type="button"
          >
            <span
              class="sr-only"
            >
              Open main menu
            </span>
            <img
              alt="Menu icon"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
          </button>
        </div>
        <a
          class="ml-5 lg:px-0"
          href="/dashboard"
        >
          <img
            alt="BCBST Logo"
            data-nimg="1"
            decoding="async"
            height="35"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
            style="color: transparent;"
            width="174"
          />
        </a>
      </div>
      <div
        class="flex items-center"
      >
        <div
          aria-label="Profile Card"
          class="flex h-full secondary-bg-color2 text-white px-4 py-1 hover:bg-info focus:bg-info focus:mr-1 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary focus:h-[90%] "
          tabindex="0"
        >
          <img
            alt="Profile Icon"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <div
            class="hidden lg:block p-2"
          >
            <span
              class="text-xs"
            >
              My Profile:
            </span>
            <p>
              Alpha Beta
            </p>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
